src/huggingface_hub/repository.py

```diff
@@ -869,14 +869,119 @@ def git_checkout(self, revision, create_branch_ok=False):
+            return False
+
+        try:
+            git_tags = subprocess.run(
+                ["git", "tag", "-d", tag_name],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                check=True,
+                encoding="utf-8",
+                cwd=self.local_dir,
+            ).stdout.strip()
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+        try:
+            git_tags = subprocess.run(
+                ["git", "push", "origin", "--delete", tag_name],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                check=True,
+                encoding="utf-8",
+                cwd=self.local_dir,
+            ).stdout.strip()
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+        return True
+
+    def add_tag(self, tag_name: str, message: str):
+        """
+        Add a tag at the current head and push it
+        """
+        try:
+            git_status = subprocess.run(
+                ["git", "tag", "-a", tag_name, "-m", message],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                check=True,
+                encoding="utf-8",
+                cwd=self.local_dir,
+            ).stdout.strip()
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+        try:
+            git_status = subprocess.run(
+                ["git", "push", "origin", tag_name],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                check=True,
+                encoding="utf-8",
+                cwd=self.local_dir,
+            ).stdout.strip()
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+    def is_repo_clean(self) -> bool:
+        """
+        Return whether or not the git status is clean or not
+        """
+        try:
+            git_status = subprocess.run(
+                ["git", "status", "--porcelain"],
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                check=True,
+                encoding="utf-8",
+                cwd=self.local_dir,
+            ).stdout.strip()
+        except subprocess.CalledProcessError as exc:
+            raise EnvironmentError(exc.stderr)
+
+        return len(git_status) == 0
+
+    def push_to_hub(
+        self, commit_message: str = "commit files to HF hub", clean_ok: bool = False
+    ) -> str:
         """
         Helper to add, commit, and push files to remote repository on the HuggingFace Hub.
         Will automatically track large files (>10MB).
 
         Args:
             commit_message: commit message.
         """
+        if clean_ok and self.is_repo_clean():
+            logger.info("Repo currently clean.  Ignoring push_to_hub")
```

Comment:
Did you run into issues with a git repository that has nothing to commit?

Reply:
Yes, that would especially happen when programmatically adding ".zip" files to a repo that already had ".zip" as part of lfs, for example